### ISSUE TITLE
Installation on Ubuntu: Make error [node-version]

### ISSUE THREAD

**Issue by v3rse** [Automattic/wp-calypso#2869]
Attempting the installation steps. I cloned the repo and I've been trying to run `make run` but so far I keep getting:

`/bin/sh: 1: Syntax error: "(" unexpected
make: *** [node-version] Error 2
`

Anyone have any idea why this is?

OS version: Ubuntu 14.04 LTS

Screenshots:
![selection_019](https://cloud.githubusercontent.com/assets/203123/12634121/ba85afd6-c56e-11e5-95df-e929e96b0b12.png)
:

**Comment by richardmtl** [comment]
Note to say that @v3rse wrote to us in ticket 2512290-t as well.

**Comment by lancewillett** [comment]
Possibly related to #2573 — see also troubleshooting steps: https://github.com/Automattic/wp-calypso/blob/master/docs/troubleshooting.md

**Comment by klimeryk** [comment]
@v3rse: please see my suggestion in #2573 - depending on your set up, you might be missing executable permissions on the `check-node-version` script. Please try `chmod +x bin/check-node-version` and verify with `ls -al bin/check-node-version`:
```
-rwxr-xr-x 1 klimeryk staff 379 Feb  7 17:37 bin/check-node-version
```
If this won't help, please post some more details about your setup:
 - what shell are you using (`bash`, `zsh`?)
 - what filesystem and what mount options (see the output for `mount`)
 - which version of `node` and `npm`

**Comment by v3rse** [comment]
Thanks for the replies guys. Sorry for the delayed response though.

@klimeryk I tried changing the permissions of the ```check-node-version``` file. Same error message. Don't think it's the same issue as in #2573 .

Find the response to your questions inline:

- **what shell are you using (bash, zsh?)**
Using `zsh`

- **what filesystem and what mount options (see the output for mount)**
`/dev/sda7 on / type ext4 (rw,errors=remount-ro)`

- **which version of node and npm**
```
node --version:
v4.2.6
npm --version
2.14.12

```
NB: Using **n** (https://github.com/tj/n) for node version management .

**Comment by klimeryk** [comment]
Hmm, I've tried with `zsh` and it's working for me.
OK, let's start from the beginning:

* please pull in the newest `master` from GitHub (`git pull origin master`)
* verify that you don't have any local modifications (`git status`)
* verify that the the `check-node-version` script has the correct permissions: `ls -l bin/check-node-version`: `-rwxr-xr-x 1 klimeryk klimeryk 380 Feb 16 00:20 bin/check-node-version`
* verify that `/usr/bin/env` can find your node installation (`/usr/bin/env node --version`) - **maybe `n` is messing up something here?**
* run `export SEMVER_GLOBAL_PATH=`npm -g root`/npm/node_modules/semver` (this variable is needed by the `check-node-version` script
* run the script manually from the root of the calypso repo: `./bin/check-node-version` - it should return status `0` (you can verify that by running `echo $?` directly after the script)
* if invoking the script directly results in the same error as previously, try invoking it explicitly via node (`node bin/check-node-version`)

Sorry for all the steps, but I wanted to cover as many bases as possible at once :grin:

**Comment by v3rse** [comment]
Thanks guys for all the help.

Finally figured it out when I encountered the same error  in another project.

The issue is with **parentheses** in my file paths. eg. `Gumballs(JS)`. It seems they weren't being escaped for some reason.

This causes the shell to spit out the `Syntax error: "(" unexpected ` message.

I've proceed to change some directory names. Putting this here for posterity sake.

**Comment by klimeryk** [comment]
Thanks for the extra information, @v3rse! It so happens that I've just merged #3583 that should fix your issues with special characters in paths :) (though please note that hot reloading might not work :/)
